pkg/prometheus/asset_store.go

```diff
@@ -109,59 +112,72 @@ func assetKeyFunc(obj interface{}) (string, error) {
 
 // addSafeTLSConfig processes the given SafeTLSConfig and adds the referenced CA, certificate and key to the store.
 func (a *assetStore) addSafeTLSConfig(ctx context.Context, ns string, tlsConfig monitoringv1.SafeTLSConfig) error {
-	if tlsConfig.CA != (monitoringv1.SecretOrConfigMap{}) {
-		var (
-			ca  string
-			err error
-		)
-
-		switch {
-		case tlsConfig.CA.Secret != nil:
-			ca, err = a.getSecretKey(ctx, ns, *tlsConfig.CA.Secret)
+	err := tlsConfig.Validate()
+	if err != nil {
+		return err
```

Comment:
nit: let's wrap the error, i think we are using the `pkg/errors` package here, so we can use something like `return errors.Wrap(err, "TLS configuration validation failed")`.